### PR TITLE
updated Form validation

### PR DESCRIPTION
**Pull request by venurockss**
The issue regarding form validation for the user registration/submission form has been successfully resolved. Previously, the form was not validating inputs properly, leading to submission of incorrect or incomplete data.

### CODE REVIEW

**Comment by Vignesh025** [comment]
There are still conflicting files in your PR. Please check and sync your repo with the latest changes before proceeding.

**Comment by venurockss** [comment]
@Vignesh025 Done with the conflicts.

File: views/users/register.ejs [hunk 1]
```diff
@@ -1,53 +1,52 @@
...
-        <script>
-            const password = document.getElementById('password');
-            const togglePassword = document.getElementById('togglePassword');
-
-            togglePassword.addEventListener('click', function () {
-                const type = password.getAttribute('type') === 'password' ? 'text' : 'password';
-                password.setAttribute('type', type);
-                this.src = type === 'password' ? "/icons/hidden.png" : "/icons/visible.png";
-            });
-        </script>
```
Comment (Sowmiyajayakumar) [inline views/users/register.ejs:53]: Undo this deletion

Comment (venurockss) [reply views/users/register.ejs:53]: Okay

File: views/users/register.ejs [hunk 2]
```diff
@@ -1,53 +1,62 @@
...
+                            <div class="valid-feedback">
+                                Looks good!
                             </div>
-                            <div class="mb-3">
-                                <label class="form-label" for="email">Email</label>
-                                <input class="form-control" type="email" id="email" name="email" required>
-                                <div class="valid-feedback">
-                                    Looks good!
-                                </div>
-                                <div class="mb-3  position-relative">
```
Comment (Sowmiyajayakumar) [inline views/users/register.ejs:25]: position-relative class is required please add it.

File: views/users/register.ejs [hunk 3]
```diff
@@ -1,53 +1,62 @@
...
-                                    <label class="form-label" for="password">Password</label>
-                                    <input class="form-control" type="password" id="password" name="password" required>
-                                    <img id="togglePassword" src="/icons/hidden.png"
-                                        class="position-absolute toggle-icon" alt="Toggle visibility" />
```
Comment (Sowmiyajayakumar) [inline views/users/register.ejs:29]: undo this deletion as well

File: views/users/register.ejs [hunk 4]
```diff
@@ -1,53 +1,62 @@
...
+                            <div class="invalid-feedback">
+                                Please enter a valid Email address 
+                            </div>
+                        </div>
+                        <div class="mb-3">
+                            <label class="form-label" for="password">Password</label>
+                          
+                            <input class="form-control" type="password" id="password" name="password" required
+                                pattern="(?=.*[A-Z])(?=(.*\d){2,}).{6,}" 
+                                title="Password must be at least 6 characters long, contain at least two digits, and one uppercase letter.">
```
Comment (Sowmiyajayakumar) [inline views/users/register.ejs:35]: title not required

Comment (venurockss) [reply views/users/register.ejs:35]: Okay

**Comment by Sowmiyajayakumar** [comment]
Hey @venurockss, this PR is outdated. Register page UI is revamped. Can you add your validation changes by creating new PR from latest main branch?

**Comment by venurockss** [comment]
Okay

**Pull request closed without merging**